### PR TITLE
don't require privileges

### PR DESCRIPTION
**Pull request by mgaunard**
This changeset makes it so that exact-capture does not attempt to perform privileged operations (ioctl SIOCSIFFLAGS and ethtool_set_priv_flags) if the settings are already correct.

This makes it possible to run exact-capture without the ability to change those parameters.

### CODE REVIEW

File: src/exactio/exactio_exanic.c [hunk 1]
```diff
@@ -231,15 +231,24 @@ static int set_exanic_params(exanic_t *exanic, char* device, int port_number,
...
     ifr.ifr_flags |= IFF_UP;
 
-    if (ioctl(fd, SIOCSIFFLAGS, &ifr) == -1)
+    if (ifr_changed)
     {
-        fprintf(stderr, "ioctl(SIOCSIFFLAGS): %s:%d: %s\n", device, port_number, strerror(errno));
-        exit(1);
+        if (ioctl(fd, SIOCSIFFLAGS, &ifr) == -1)
+        {
+            fprintf(stderr, "ioctl(SIOCSIFFLAGS): %s:%d: %s\n", device, port_number, strerror(errno));
```
Comment (mgrosvenor) [inline src/exactio/exactio_exanic.c]: I really like that you've made these errors more explicit, but can I ask for two quick changes before I merge in:
1) can you please replace fprintf() with the libchaste logging functions ch_log_XXX which are used throughout the project? I'd suggest ch_log_error() or ch_log_fatal() in this instance. See https://github.com/exablaze-oss/exact-capture/blob/master/libs/libchaste/include/log/log.h
2) Can you make the error text more descriptive? See https://github.com/exablaze-oss/exact-capture/blob/89615ae0f9ceed8a7ef5df78a1189dec32cb924b/src/exact-capture-listener.c#L449

Thanks!

File: src/exactio/exactio_exanic.c [hunk 2]
```diff
@@ -231,15 +231,24 @@ static int set_exanic_params(exanic_t *exanic, char* device, int port_number,
...
+            exit(1);
```
Comment (mgrosvenor) [inline src/exactio/exactio_exanic.c]: Be careful with a simple exit here. There's a whole chain of dependencies that lead us to this function, so we're probably deep inside a thread construction block. Can I suggest a return statement here. It should cascade upwards correctly.

**Comment by mgrosvenor** [comment]
I'm sorry it took me so long to get around to looking a this. For some reason I never got a notification your pull request. Apologies! 

I really like the changes you've proposed. I can see that what you've done is completely consistent with the file in which you found it. However, it looks like the contents of this file was copy-pasted from somewhere else, and hence is out of sync with the rest of the code base. That changes I've proposed would bring it into line with the rest of the code base. Sorry to be a hassle, I know you didn't do anything wrong here, but at the same time, I don't want to add more "wrong" to the code than we already have.

File: src/exactio/exactio_exanic.c (hunk 2 above)
Comment (mgaunard) [reply src/exactio/exactio_exanic.c]: There is already an exit in this function which I did not add.
Do you want all of them to be changed to return 1?

**Review by mgrosvenor — APPROVED** [review]

**Review by mgrosvenor — APPROVED** [review]

**Comment by mgrosvenor** [comment]
Thanks for your PR. I've tided up the code and fixed the merge conflict. It should now do what you were looking for. 

Cheers,
M

**Pull request merged**